src/network.lisp

```diff
@@ -17,17 +17,24 @@
              (format stream "Request error: ~A" (what condition)))))
 
 
-(defun make-request (bot name options &key (streamp nil) (timeout 3))
+(defun make-request (bot name &rest options &key (streamp nil) (timeout 3) &allow-other-keys)
   "Perform HTTP request to 'name API method with 'options JSON-encoded object."
   (let ((url (concatenate 'string (get-endpoint bot) name)))
     (log:debug "Posting data to"
                (obfuscate url)
                options)
     (let* ((max-timeout (* timeout 10))
+           (processed-options (loop for (key value)
+                                      on (alexandria:remove-from-plistf options :timeout :streamp)
```

Comment:
`remove-from-plistf` changes the place like `setf`. Here in the loop you probably don't need it.

Please, replace it with `remove-from-plist` which returns a new list. It is always better to avoid data modifications when possible.

After this small change, we'll merge the pull.

It would be nice to get rid of some `nconc` used to create `options` lists in other places though. But this could be made in a separate pull.

Reply:
Done!